### PR TITLE
Refactor prediction input model

### PR DESCRIPTION
**Pull request by vin-spiegel**
I think there is no interface about 'PredictionInput' to use varient inputs.
I did
- Added 'IPredictionInput'
- Refactor 'PredictionInput' to 'DefaultPredictionInput'
- Added NegativePrompt in DefaultPredictionInput

### CODE REVIEW

**Comment by StefH** [comment]
@vin-spiegel 
Thank you.

I'll take a look.

File: examples/Replicate.Net.Common/Example/Client/IExampleApi.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public interface IExampleApi
     Task<Prediction> CancelPredictionAsync([Path] string predictionId, CancellationToken cancellationToken = default);
 
     [Post("predictions")]
-    Task<Prediction> CreatePredictionAsync([Body] PredictionInput input, CancellationToken cancellationToken = default);
+    Task<Prediction> CreatePredictionAsync([Body] DefaultPredictionInput input, CancellationToken cancellationToken = default);
```
Comment (StefH) [inline examples/Replicate.Net.Common/Example/Client/IExampleApi.cs]: Should this be the interface?

File: examples/Replicate.Net.Common/Example/Client/IExampleApiExtensions.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace Replicate.Net.Common.Example.Client;
 
 public static class IExampleApiExtensions
 {
-    public static async Task<Prediction> CreatePredictionAndWaitOnResultAsync(this IExampleApi api, PredictionInput input, int timeoutInSeconds = 5 * 60, CancellationToken cancellationToken = default)
+    public static async Task<Prediction> CreatePredictionAndWaitOnResultAsync(this IExampleApi api, DefaultPredictionInput input, int timeoutInSeconds = 5 * 60, CancellationToken cancellationToken = default)
```
Comment (StefH) [inline examples/Replicate.Net.Common/Example/Client/IExampleApiExtensions.cs]: Should this be the interface?

File: src/Replicate.Net/Models/DefaultPredictionInput.cs [hunk 3]
```diff
@@ -1,17 +1,23 @@
 ﻿using Newtonsoft.Json;
+using Replicate.Net.Interop;
 
 namespace Replicate.Net.Models;
 
 /// <summary>
 /// The model's input.
```
Comment (StefH) [inline src/Replicate.Net/Models/DefaultPredictionInput.cs]: The default model's input.

File: src/Replicate.Net/Interop/IPredictionInput.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace Replicate.Net.Interop;
+
```
Comment (StefH) [inline src/Replicate.Net/Interop/IPredictionInput.cs]: Add summary:
```
/// <summary>
/// The model's input.
/// </summary>
```

File: src/Replicate.Net/Interop/IPredictionInput.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+namespace Replicate.Net.Interop;
```
Comment (StefH) [inline src/Replicate.Net/Interop/IPredictionInput.cs]: Rename this to `Replicate.Net.Interfaces`

**Review by StefH — CHANGES_REQUESTED** [review]

File: examples/Replicate.Net.Common/Example/Client/IExampleApi.cs (hunk 1 above)
Comment (vin-spiegel) [reply examples/Replicate.Net.Common/Example/Client/IExampleApi.cs]: Oh yes. this should be interface

**Pull request closed without merging**